src/hpa.c

```diff
@@ -924,11 +919,8 @@ hpa_time_until_deferred_work(tsdn_t *tsdn, pai_t *self) {
 			malloc_mutex_unlock(tsdn, &shard->mtx);
 			return BACKGROUND_THREAD_DEFERRED_MIN;
 		}
-		nstime_t nstime;
-		shard->central->hooks.curtime(&nstime,
-		    /* first_reading */ true);
```

Comment:
I'm a bit worried about this now - if you need to mock time for the whole HPA, you now have to employ two mechanisms: this hook and `nstime_update`. Maybe we need to make another `nstime_ns_since_with_hook` that has a function pointer argument?

Reply:
The `nstime_update` pointer switch does feel like cheating, given that we already have the time related hooks. Let me update with another hook then.